src/extension.ts

```diff
@@ -519,30 +519,39 @@ class Device extends vscode.TreeItem {
     }
     
     async captureScreenshot() {
-        const error = (message) => {
-            vscode.window.showErrorMessage("Error capturing screenshot: " + message);
+        let anyErrors = false;
+        const handleCaptureError = (e) => {
+            anyErrors = true;
+            vscode.window.showErrorMessage("Error capturing screenshot: " + (e.message || e));
         }
-        
+
         try {
             const screenshotDirectory = await getSharedTempDir('ev3dev-screenshots');
             const screenshotFile = `${screenshotDirectory}/ev3dev-${sanitizedDateString()}.png`;
 
             const conn = await this.exec('fbgrab -');
             const writeStream = fs.createWriteStream(screenshotFile);
+
+            conn.on('error', (e: Error) => handleCaptureError);
+
             writeStream.on('open', () => {
                 conn.stdout.pipe(writeStream);
             });
 
             writeStream.on('close', () => {
-                vscode.window.setStatusBarMessage(`Screenshot successfully captured`);
-                vscode.commands.executeCommand('vscode.open', vscode.Uri.file(screenshotFile));
+                if(!anyErrors) {

```

Comment:
🤔 It's been a while since the last time I worked with streams in detail and they've changed a bit since then, so my recollection is a bit fuzzy. I believe that you can still flush all data and close the stream even if errors were raised, as the error event can be raised arbitrarily by the stream source. I couldn't find any documentation which made me confident that there was an event for "finished and no errors were thrown", but I'd be happy to switch and remove the ugly flag if we can find such a thing.